File: tools-manage/invocationlab-admin/src/main/java/org/springframework/cloud/openfeign/ApiJarResourceLoader.java
package org.springframework.cloud.openfeign;

import org.javamaster.invocationlab.admin.service.load.classloader.ApiJarClassLoader;
import org.jetbrains.annotations.NotNull;
import org.springframework.core.io.Resource;
import org.springframework.core.io.ResourceLoader;

/**
 * @author yudong
 * @date 2022/11/9
 */
public class ApiJarResourceLoader implements ResourceLoader {
    private final ApiJarClassLoader apiJarClassLoader;

    public ApiJarResourceLoader(ApiJarClassLoader apiJarClassLoader) {
        this.apiJarClassLoader = apiJarClassLoader;
    }

    @Override
    public @NotNull Resource getResource(@NotNull String location) {
        throw new UnsupportedOperationException();
    }

    @Override
    public ClassLoader getClassLoader() {
        return apiJarClassLoader;
    }
}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/springframework/cloud/openfeign/ApiJarResourceLoader.kt
package org.springframework.cloud.openfeign

import org.javamaster.invocationlab.admin.service.load.classloader.ApiJarClassLoader
import org.springframework.core.io.Resource
import org.springframework.core.io.ResourceLoader

/**
 * @author yudong
 * @date 2022/11/9
 */
class ApiJarResourceLoader(private val apiJarClassLoader: ApiJarClassLoader) : ResourceLoader {
    override fun getResource(location: String): Resource {
        throw UnsupportedOperationException()
    }

    override fun getClassLoader(): ClassLoader {
        return apiJarClassLoader
    }
}
